src/content/docs/en/reference/adapter-reference.mdx

```diff
@@ -197,10 +211,21 @@ const response = await app.render(request);
 
 ##### `RenderOptions`
 
+<p>
+
+**Type:** `{addCookieHeader?: boolean; clientAddress?: string; locals?: object; routeData?: RouteData;}`
```

Comment:
You can [check the type here](https://github.com/withastro/astro/blob/a2f8c5d85ff15803f5cedf9148cd70ffc138ddef/packages/astro/src/core/app/index.ts#L30).